### PR TITLE
PostStore.SaveMultiple with Team

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Change the signature of `PostStore.SaveMultiple` to expect a `TeamID`, allowing that code to avoid assuming posts have channels that can supply that team metadata. (This assumption won't be true once we support topical threads.)

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-48022

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/import_functions.go [hunk 1]
```diff
@@ -1176,7 +1176,7 @@ func (a *App) importReplies(c request.CTX, data []imports.ReplyImportData, post
 	}
 
 	if len(postsForCreateList) > 0 {
-		if _, _, err := a.Srv().Store().Post().SaveMultiple(postsForCreateList); err != nil {
+		if _, _, err := a.Srv().Store().Post().SaveMultiple(teamID, postsForCreateList); err != nil {
```
Comment (lieut-data) [inline app/import_functions.go:1179]: We are conveniently importing posts all from the same team in this context.

File: app/import_functions.go [hunk 2]
```diff
@@ -1467,8 +1467,8 @@ func (a *App) importMultiplePostLines(c request.CTX, lines []imports.LineImportW
 		postsWithData = append(postsWithData, postAndData{post: post, postData: line.Post, team: team, lineNumber: line.LineNumber})
 	}
 
-	if len(postsForCreateList) > 0 {
-		if _, idx, nErr := a.Srv().Store().Post().SaveMultiple(postsForCreateList); nErr != nil {
+	for teamID, postsForCreateList := range postsForCreateByTeam {
+		if _, idx, nErr := a.Srv().Store().Post().SaveMultiple(teamID, postsForCreateList); nErr != nil {
```
Comment (lieut-data) [inline app/import_functions.go:1471]: We now batch the posts by team, but otherwise import normally.

File: store/sqlstore/post_store.go [hunk 3]
```diff
@@ -271,7 +271,24 @@ func (s *SqlPostStore) SaveMultiple(posts []*model.Post) ([]*model.Post, int, er
...
+
+	posts, _, err := s.SaveMultiple(teamID, []*model.Post{post})
+	if err != nil {
+		return nil, err
+	}
+
+	return posts[0], nil
+}
+
+func (s *SqlPostStore) SaveWithTeamID(teamID string, post *model.Post) (*model.Post, error) {
```
Comment (iomodo) [inline store/sqlstore/post_store.go:290]: 👍 Let's not forget to use this in createPost instead of `Save`

**Review by iomodo — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks for the descriptive comments @lieut-data 💯

**Comment by lieut-data** [comment]
@furqanmlk, heads up that I'm not expecting to ship threads that don't belong to channels anytime soon, so I'm inclined to retire this PR, resurrecting it if needed in the future instead of adding potential churn to an upcoming release.

**Pull request closed without merging**